shoop/configuration.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from shoop.core import cache
+from shoop.core.models import Shop, Configuration
+
+
+SHOP_CONFIGURATION_CACHE_PREFIX = "shop_configuration_%s"
+
+
+def get_cache_key(shop):
+    return SHOP_CONFIGURATION_CACHE_PREFIX % shop.pk if shop else ""
+
+
+def set(shop, key, data):
+    """
+    Set shop, key, data combination to configurations.
+    After configuration create or update bump cache version
+    or invalidate cache for all shops
+
+    :param shop: shop for configuration
+    :param key: configuration key
+    :param data: data set to configuration
+    """
+    conf, _ = Configuration.objects.get_or_create(shop=shop, key=key)
+    conf.data = data
+    conf.save()
+
+    if shop:
+        cache.bump_version(get_cache_key(shop))
```

Comment:
Uhm, I think you've misunderstood how the versioning API works. It works for key _prefixes_ (namespaces), not full keys.

You'll really just want to do
```python
cache.set(get_cache_key(shop), None)
```
here, _or_ refactor the code to have a common namespace (`shop_config` sounds good), so the actual cache keys are of the format `shop_config:123` (for shop id 123), after which you can do
```python
if shop:
    cache.set(get_cache_key(shop), None)  # single-shop setting changed
else:
    bump_version("shop_config")  # global setting changed
```
instead of having to iterate through all shops to invalidate the global setting layer.